glib/src/translate.rs

```diff
@@ -604,6 +604,14 @@ impl<'a> ToGlibPtr<'a, *const c_char> for Path {
         let tmp = path_to_c(self);
         Stash(tmp.as_ptr(), tmp)
     }
+
+    #[inline]
+    fn to_glib_full(&self) -> *const c_char {
+        let stash = self.to_glib_none();
+        unsafe {
+            ffi::g_strdup(stash.0) as *mut c_char
```

Comment:
Since `to_glib_none` allocates anyway, it seems like it'd make more sense to implement `to_glib_none` in terms of `to_glib_full`, and not the other way around, potentially saving an allocation.

Reply:
Good catch, that's true (also for all the other string-y types). The main problem is that we then have to replicate the `NUL`-check of `CString`, or silently truncate at `NUL`. For now continuing with the current behaviour should be safest (and I think is also the correct behaviour here).

Do you want to add another commit on top of this to implement this?

Reply:
Well there's also the Windows special case here: in GLib they're all UTF-8 but in Rust [they're](https://doc.rust-lang.org/std/ffi/struct.OsString.html)

> On Windows, strings are often arbitrary sequences of non-zero 16-bit values, interpreted as UTF-16 when it is valid to do so.

So that conversion also needs to be moved around a bit.